packages/serialized-state-manager/src/internal/serialized-state-manager.ts

```diff
@@ -1,19 +1,27 @@
 import {SerializedStateManagerV0} from '..';
+import {ClientSideStateManager} from './client-side-state-manager';
+import {ServerSideStateManager} from './server-side-state-manager';
 
 export class SerializedStateManager implements SerializedStateManagerV0 {
-  public register(_createSerializedState: () => string | undefined): void {
-    throw new Error('Method not implemented.');
+  public constructor(
+    private readonly consumerUid: string,
+    private readonly serverSideStateManager: ServerSideStateManager,
```

Comment:
Based on the implementation I can't see why we need these two client/server classes. I would implement it directly here and not use delegation...